src/symbolize/gimli/libs_windows.rs

```diff
@@ -43,13 +42,79 @@ unsafe fn add_loaded_images(ret: &mut Vec<Library>) {
     }
 }
 
+// Safety: long_path should be null-terminated
+#[cfg(target_os = "cygwin")]
+unsafe fn get_posix_path(long_path: &[u16]) -> Option<OsString> {
+    use super::mystd::os::unix::ffi::OsStringExt;
+
+    unsafe extern "C" {
+        // Doc: https://cygwin.com/cygwin-api/func-cygwin-conv-path.html
+        // Src: https://github.com/cygwin/cygwin/blob/718a15ba50e0d01c79800bd658c2477f9a603540/winsup/cygwin/path.cc#L3902
+        // Safety:
+        // * `what` should be `CCP_WIN_W_TO_POSIX` here
+        // * `from` is `*const u16`, null-terminated, UTF-16 path
+        // * `to` is `*mut u8` buffer, the buffer size is `size`.
+        fn cygwin_conv_path(
+            what: libc::c_uint,
+            from: *const libc::c_void,
+            to: *mut libc::c_void,
+            size: libc::size_t,
+        ) -> libc::ssize_t;
+    }
+    const CCP_WIN_W_TO_POSIX: libc::c_uint = 3;
+
+    // If `size` is 0, returns needed buffer size, including null terminator;
+    // or -1 if error.
+    // Safety: **Confirmed from source:** If `size` is 0, `to` is not used.
+    let name_len = unsafe {
+        cygwin_conv_path(
+            CCP_WIN_W_TO_POSIX,
+            long_path.as_ptr().cast(),
+            core::ptr::null_mut(),
+            0,
+        )
+    };
+    // Expect at least 1 for null terminator.
+    // It's not likely to return error here.
+    if name_len < 1 {
+        return None;
+    }
+    let name_len = name_len as usize;
+    let mut name_buffer = vec![0_u8; name_len];
+    // Safety: `name_buffer` is large enough.
+    let res = unsafe {
+        cygwin_conv_path(
+            CCP_WIN_W_TO_POSIX,
+            long_path.as_ptr().cast(),
+            name_buffer.as_mut_ptr().cast(),
+            name_buffer.len(),
+        )
+    };
+    // It's not likely to return error here.
+    if res != 0 {
+        return None;
+    }
+    // Ignore null terminator.
+    unsafe { name_buffer.set_len(name_len - 1) };
+    let name = OsString::from_vec(name_buffer);
```

Comment:
...I wonder...

I don't believe an OsStr actually has to not include `'\0'` in its span? So we could just not bother updating the len here, if we want.

Reply:
OK.

Reply:
Well, well... it seems that I *should* set the length. If I don't, the code doesn't work.

Reply:
`mmap` opens the file, while the path passed to `File::open` could not contain null bytes.
```
Error { kind: InvalidInput, message: "file name contained an unexpected NUL byte" }
```